Support fetching two letter chains

pslacerda opened the pull request:
The structure 4WQF has chains with two letter codes. This commit allow `fetch` to import such chains (eg. `fetch 4WQF.BZ`).

speleo3 commented:
Looks fine. One question: is `.` a common separator for PDB code and chain? I have seen others (like `:` or `_`) but not `.`.

pslacerda commented:
Just added these separators, seems to work fine. Yes, `.` it is valid, although  is a bit annoying because the newly created object name isn't standardized (ie. converted to `_` as the others).

speleo3 commented:
Thanks for the update. I think we should do it like this:

```python
            chain = None
            if type in ('pdb', 'cif', 'mmtf'):
                m = re.match(r'(\d[0-9a-z]{3})[:_.]?([a-z]+)$', obj_code, flags=re.I)
                if m is not None:
                    obj_code, chain = m.group(1), m.group(2)
```

pslacerda commented:
Never know when 5 character PDB ids will come along.

speleo3 commented:
8 character PDB codes are already there:
https://www.rcsb.org/news?year=2017&article=59cd3fa822e913222ac88fcd

pslacerda commented:
Nice update.

> 8 character PDB codes are already there:
> https://www.rcsb.org/news?year=2017&article=59cd3fa822e913222ac88fcd

There is any id that was already replaced by the new standard? If the updates happens as the article said, old accession codes will increasingly be obsoleted.

speleo3 commented:
> There is any id that was already replaced by the new standard?

I don't think so.
